### ISSUE TITLE
Dear Team after training model evaluation precision is nan for lots of catedories and some of having 0 how could i solve it because i am using faster rcnn resnet 101 model

### ISSUE THREAD

**Issue by DeveloperRachit** [tensorflow/tensorflow#34699]
Dear Team after training model evaluation precision is nan for lots of catedories and some of having 0 how could i solve it because i am using faster rcnn resnet 101 model

**Comment by oanush** [comment]
@DeveloperRachit ,
Can you share a simple and standalone code to reproduce the issue? also mention Tensorflow version being used.Thanks!

**Comment by DeveloperRachit** [comment]
yes

**Comment by DeveloperRachit** [comment]
this is my config file faster_rcnn_resnet101.config"

**Comment by DeveloperRachit** [comment]
model {
  faster_rcnn {
    num_classes: 545
    image_resizer {
      fixed_shape_resizer {
        height: 350
        width: 350
      }
    }
    feature_extractor {
      type: 'faster_rcnn_resnet101'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
 first_stage_nms_score_threshold: 0.1
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: true
        dropout_keep_probability: 0.5
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
                                                                                                                            29,5          32%
  second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.009999999776482582
        iou_threshold: 0.6000000238418579
        max_detections_per_class: 100
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}

train_config: {
  batch_size: 2
    optimizer {
        adam_optimizer: {
            learning_rate {
            exponential_decay_learning_rate: {initial_learning_rate:0.0001}
            }
        }
    }
  fine_tune_checkpoint: "/data/sample-apps/deep_dive_demos/open_images_detection/preprocessing/faster_rcnn_resnet101_coco_2018_01_28/model.ckpt"
  from_detection_checkpoint: true
  batch_queue_capacity: 50
  gradient_clipping_by_norm: 10
  data_augmentation_options {
    random_horizontal_flip {
                                   }
  }
}

eval_config: {
  num_examples:2000
  num_visualizations: 20

}

train_input_reader: {
  tf_record_input_reader {
    input_path: "/data/sample-apps/deep_dive_demos/open_images_detection/preprocessing/record.train"
  }
  label_map_path: "/data/sample-apps/deep_dive_demos/open_images_detection/preprocessing/models/research/object_detection/data/oid_v4_label_map.pbtxt"
}


eval_input_reader: {
  tf_record_input_reader {
    input_path: "/data/sample-apps/deep_dive_demos/open_images_detection/preprocessing/record.test"
  }
  label_map_path: "/data/sample-apps/deep_dive_demos/open_images_detection/preprocessing/models/research/object_detection/data/oid_v4_label_map.pbtxt"
  shuffle: false
  num_readers: 2
}

**Comment by DeveloperRachit** [comment]
and rest of i am using train.py which is inside object detection api

**Comment by DeveloperRachit** [comment]
https://github.com/tensorflow/models/blob/master/research/object_detection/legacy/train.py

this is above link that i am using to train my datasets

**Comment by DeveloperRachit** [comment]
https://github.com/tensorflow/models/blob/master/research/object_detection/legacy/eval.py

**Comment by DeveloperRachit** [comment]
INFO:tensorflow:global step 258266: loss = 650309252677632.0000 (0.618 sec/step)
INFO:tensorflow:global step 258266: loss = 650309252677632.0000 (0.618 sec/step)
some time losses is very high what is exact problem 
i am using GPU Quadro M5000 with 8GB Ram 
and having 32 GB CPU RAM

**Comment by gowthamkpr** [comment]
@DeveloperRachit When TP+FP=0 : means that all instances were predicted as negative. You can try referring to the issue [here](https://github.com/tensorflow/models/issues/1621).

You can follow the tutorial thats mentioned [here1](https://github.com/vijendra1125/Tensorflow_Object_detection_API-Custom_Faster_RCNN) and [here2](https://medium.com/@vijendra1125/tensorflow-api-custom-object-detection-55444de6562d) to learn about object detection as the issue might be that your.

Please post these questions in stackoverflow as github is  only meant for bug/performance, build/install, feature request, docs related issues. Thank you!
